src/core/cli/watcher.go

```
@@ -302,7 +304,75 @@ func (aw *AgentWatcher) terminateAgent() {
+			// try to stop the process even if the .pid file lookup misfired.
+			fmt.Printf("[watch] lifecycle kill for %s reported: %v (falling back to direct signal)\n", name, err)
+		}
+	}
+
+	pid := cmd.Process.Pid
+	pgid := pid
+	_ = syscall.Kill(-pgid, syscall.SIGTERM)
+	deadline := time.Now().Add(aw.config.StopTimeout)
+	for time.Now().Before(deadline) {
+		if err := syscall.Kill(-pgid, 0); err != nil {
+			return nil
```

golangci-lint (2.11.4)

[error] error is not nil (line 347) but it returns nil

(nilerr)